src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -394,6 +394,20 @@ public Parser(Compilation compilation)
                 };
             }
 
+            private static string EscapeString(string representation)
```

Comment:
This appears kind of ad-hoc. Consider using the built-in [SytaxFactory.Literal](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.visualbasic.syntaxfactory.literal?view=roslyn-dotnet-4.6.0#microsoft-codeanalysis-visualbasic-syntaxfactory-literal(system-string)) factory to escape the string as expected.

Reply:
Please update the `FormatStringLiteral` method in the emitter class so that `SyntaxFactory` is used there:

https://github.com/dotnet/runtime/blob/45865877ecd3d1506e0aa19d810fb35bed4b5f9e/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs#L1239

Reply:
@eiriktsarpalis , I'll try to understand how to use [SytaxFactory.Literal](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.visualbasic.syntaxfactory.literal?view=roslyn-dotnet-4.6.0#microsoft-codeanalysis-visualbasic-syntaxfactory-literal(system-string)) on weekend and apply required changes.
You had provided link to Microsoft.CodeAnalysis.**VisualBasic**.SyntaxFactory.Literal. Are you sure, that it's correct? Why not Microsoft.CodeAnalysis.**CSharp**.SyntaxFactory.Literal?

Reply:
Apologies, of course it should be the `CSharp` equivalent.

Reply:
Fixed by https://github.com/dotnet/runtime/pull/84688/commits/75dedede15be085d7fb2cb3ab4a8da78acfd0514